geomesa-gt/geomesa-gt-partitioning/src/test/scala/org/locationtech/geomesa/gt/partition/postgis/PartitionedPostgisDataStoreTest.scala

```diff
@@ -0,0 +1,158 @@
+/***********************************************************************
+ * Copyright (c) 2013-2022 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.gt.partition.postgis
+
+import com.typesafe.scalalogging.LazyLogging
+import org.geotools.data.postgis.PostGISPSDialect
+import org.geotools.data.{DataStoreFinder, DefaultTransaction, Transaction}
+import org.geotools.filter.identity.FeatureIdImpl
+import org.geotools.filter.text.ecql.ECQL
+import org.geotools.jdbc.JDBCDataStore
+import org.geotools.util.factory.Hints
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.gt.partition.postgis.dialect.PartitionedPostgisDialect
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.io.WithClose
+import org.locationtech.geomesa.utils.text.WKTUtils
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+import scala.util.control.NonFatal
+
+@RunWith(classOf[JUnitRunner])
+class PartitionedPostgisDataStoreTest extends Specification with LazyLogging {
+
+  import scala.collection.JavaConverters._
+
+  val hours = 1
+  val spec =
+    "name:String,age:Int,dtg:Date,*geom:Point:srid=4326;" +
+        Seq(
+          s"pg.partitions.interval.hours=$hours",
+          "pg.partitions.cron.minute=0"/*,
+          "pg.partitions.max=2",
+          "pg.partitions.tablespace.wa=wa",
+          "pg.partitions.tablespace.wa-partitions=wapa",
+          "pg.partitions.tablespace.main=main"*/
+        ).mkString(",")
+
+  val methods =
+    Methods(
+      create = true,
+      write = true,
+      update = false,
+      delete = false,
+      remove = false
+    )
+
+  lazy val sft = SimpleFeatureTypes.createType(s"test", spec)
+
+  "PartitionedPostgisDataStore" should {
+    "work" in {
+      skipped("requires postgis instance")
```

Comment:
Whew...  I almost want to recommend TestContainers so that we could have a PostGIS unit (well, integration, really) tests here.

https://github.com/murrayk/postgis-integration-test-example seems to work out some of the details.

Reply:
mm requires docker though...

Reply:
Yeah...  I wonder if there's way to use Docker when you have it and skip it when you don't.... that's probably too finicky.

Is it worth adding explicit testing for this datastore to the minor release punch/check list?

Reply:
probably only requires explicit testing when upgrading geotools versions...

Reply:
That's true.  So we would want to test on major version changes then!  (Assuming that GeoTools version changes only happen on major version changes.)